src/resources/movies.ts

```diff
@@ -0,0 +1,275 @@
+import { Router, Request, Response } from 'express';
+import mongoose from 'mongoose';
+
+import BadRequestError from '../errors/BadRequestError';
+
+import { Movie } from '../schemas/movies';
+
+export const getAllMoviesRouter = Router();
+export const getMoviesByGenreRouter = Router();
+export const addNewMovieRouter = Router();
+export const updateMovieByTitleRouter = Router();
+export const deleteMovieByIdRouter = Router();
+
+const ObjectId = mongoose.Types.ObjectId;
+
+/**
+ * @swagger
+ * /movies:
+ *   get:
+ *     summary: Get a list of all movies
+ *     description: Retrieve a list of all movies from the database.
+ *     responses:
+ *       200:
+ *         description: List of movies retrieved successfully.
+ *         content:
+ *           application/json:
+ *             schema:
+ *               type: array
+ *               items:
+ *                 $ref: '#/components/schemas/Movie'
+ *       404:
+ *         description: No movies was found.
+ */
+getAllMoviesRouter.get('/', async (req: Request, res: Response): Promise<void> => {
+  const movies = await Movie.find().catch((err: Error) => {
+    throw new BadRequestError({code: 404, message: err.message });
+  });
+
+  res.status(200).json(movies);
+  });
+  
+  /**
+   * @swagger
+   * /movies/genres/{name}:
+   *   get:
+   *     summary: Get a list of movies by genre
+   *     description: Retrieve a list of movies by a specific genre from the database.
+   *     parameters:
+   *       - in: path
+   *         name: name
+   *         required: true
+   *         description: The name of the genre to filter movies by.
+   *         schema:
+   *           type: string
+   *     responses:
+   *       200:
+   *         description: List of movies retrieved successfully.
+   *         content:
+   *           application/json:
+   *             schema:
+   *               type: array
+   *               items:
+   *                 $ref: '#/components/schemas/Movie'
+   *       404:
+   *         description: Movie with title was not found.
+   */
+  getMoviesByGenreRouter.get('/:name', async (req: Request, res: Response): Promise<void> => {
+    const { name } = req.params;
+
+    const movies = await Movie.find({ genre: name })
+    .then((movies) => {
+      return movies;
+    })
+    .catch((err: Error) => {
+      throw new BadRequestError({code: 404, message: err.message });
+    });
+    
+        res.status(200).json(movies);
+  });
+  
+  
+  /**
+   * @swagger
+   * /movies/{title}:
+   *   post:
+   *     summary: Add a new movie
+   *     description: Add a new movie to the database.
+   *     parameters:
+   *       - in: path
+   *         name: title
+   *         required: true
+   *         description: The title of the movie.
+   *         schema:
+   *           type: string
```

Comment:
The same as for genres, seems redundant for POST endpoint

Reply:
I didn't quite understand it. Delete it everywhere or only where you marked. Where noted, I deleted

Reply:
done